### PR TITLE
JSDK-3564 - Adding Remote view in screen share example

### PR DESCRIPTION
**Pull request by PikaJoyce**
JIRA Ticket : [JSDK-3564](https://issues.corp.twilio.com/browse/VIDEO-3564)

In this PR I've added the remote participant view and adjusted the helper function a bit since I need access to the Media Stream in order to catch the `onended` event from when we stop sharing out of view. I have also gone ahead and updated the Twilio-Video package to use 2.11.0.

ScreenShare

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: examples/screenshare/src/index.js [hunk 1]
```diff
@@ -1,46 +1,128 @@
...
+    }
+  };
 }());
 
 function toggleButtons() {
   captureScreen.style.display = captureScreen.style.display === 'none' ? '' : 'none';
   stopScreenCapture.style.display = stopScreenCapture.style.display === 'none' ? '' : 'none';
 }
+
+function trackPublish(publishType, publication, view) {
```
Comment (charliesantos) [inline examples/screenshare/src/index.js:110]: This is an event handler right? Not sure what is the convention already used in quickstart, but it's better to use something like `onTrackPublished` or `trackPublishHandler`.

Comment (PikaJoyce) [reply examples/screenshare/src/index.js]: Good point! Thanks Charlie, it's been updated :)

**Review by charliesantos — APPROVED** [review]

File: examples/screenshare/src/helpers.js [hunk 2]
```diff
@@ -20,9 +20,7 @@ function createScreenTrack(height, width) {
       height: height,
       width: width
     }
-  }).then(function(stream) {
-    return new Video.LocalVideoTrack(stream.getVideoTracks()[0]);
-  });
```
Comment (manjeshbhargav) [inline examples/screenshare/src/helpers.js:25]: You don't need to change this function in order to listen to `onended`. You can just do it like this:
```js
const screenTrack = await createScreenTrack(...);
screenTrack.mediaStreamTrack.onended = stopScreenSharing;
```

File: examples/screenshare/src/index.js [hunk 3]
```diff
@@ -1,46 +1,128 @@
...
+    }
+  };
 }());
 
 function toggleButtons() {
   captureScreen.style.display = captureScreen.style.display === 'none' ? '' : 'none';
   stopScreenCapture.style.display = stopScreenCapture.style.display === 'none' ? '' : 'none';
 }
+
+function trackPublishedHandler(publishType, publication, view) {
```
Comment (manjeshbhargav) [inline examples/screenshare/src/index.js:110]:
```suggestion
function onTrackPublished(publishType, publication, view) {
```

**Review by manjeshbhargav — CHANGES_REQUESTED** [review]

File: examples/screenshare/src/helpers.js (hunk 2 above)
Comment (PikaJoyce) [reply examples/screenshare/src/helpers.js]: Thanks Manjesh! It is done

File: examples/screenshare/src/helpers.js [hunk 4]
```diff
@@ -3,8 +3,8 @@
 const Video = require('twilio-video');
 
 /**
- * Create a LocalVideoTrack for your screen. You can then share it
- * with other Participants in the Room.
+ * Create a screen track for your screen. You can then publish it
+ * to other Participants in the Room.
```
Comment (manjeshbhargav) [inline examples/screenshare/src/helpers.js:7]: @PikaJoyce You can revert this change.

File: examples/screenshare/src/index.js [hunk 5]
```diff
@@ -1,46 +1,127 @@
...
+    name: roomLocal.name,
+    tracks: []
+  });
+
   // Hide the "Stop Capture Screen" button.
   stopScreenCapture.style.display = 'none';
 
   // The LocalVideoTrack for your screen.
   let screenTrack;
+  let localScreenTrack;
```
Comment (manjeshbhargav) [inline examples/screenshare/src/index.js:41]:
```suggestion
```

File: examples/screenshare/src/index.js [hunk 6]
```diff
@@ -1,46 +1,127 @@
...
-      screenTrack.attach(screenPreview);
+
+      // Publish screen track to room
+      await roomLocal.localParticipant.publishTrack(screenTrack, {
+        priority: 'low',
+      }).then (trackPublication => {
+        localScreenTrack = trackPublication.track;
+        trackPublication.track.attach(screenPreview);
+      });
+
```
Comment (manjeshbhargav) [inline examples/screenshare/src/index.js:55]:
```suggestion
      screenTrack.attach(screenPreview);

      // Publish screen track to room
      await roomLocal.localParticipant.publishTrack(screenTrack);
```

File: examples/screenshare/src/index.js [hunk 7]
```diff
@@ -1,46 +1,127 @@
...
       // Show the "Capture Screen" button after screen capture stops.
-      screenTrack.on('stopped', toggleButtons);
+      localScreenTrack.on('stopped', () => {
+        stopScreenSharing
+      });
+
+      // Listening to on ended event on the MediaStreamTrack.
+      screenTrack.mediaStreamTrack.onended = stopScreenSharing;
+
```
Comment (manjeshbhargav) [inline examples/screenshare/src/index.js:63]:
```suggestion
      // When screen sharing is stopped, unpublish the screen track.
      screenTrack.on('stopped', () => {
        if (roomLocal) {
          roomLocal.localParticipant.unpublishTrack(screenTrack);
        }
        toggleButtons();
      });

```

File: examples/screenshare/src/index.js [hunk 8]
```diff
@@ -1,46 +1,127 @@
...
     } catch (e) {
       alert(e.message);
     }
   };
 
-  stopScreenCapture.onclick = function() {
-    // Stop capturing your screen.
+  // Stop capturing your screen.
+  const stopScreenSharing = () => {
+    roomLocal.localParticipant.unpublishTrack(screenTrack);
```
Comment (manjeshbhargav) [inline examples/screenshare/src/index.js:73]:
```suggestion
  // Stop capturing your screen.
  const stopScreenSharing = () => {
```

File: examples/screenshare/src/index.js [hunk 9]
```diff
@@ -1,46 +1,127 @@
...
     screenTrack.stop();
+    toggleButtons();
```
Comment (manjeshbhargav) [inline examples/screenshare/src/index.js:75]:
```suggestion
```

File: examples/screenshare/src/index.js [hunk 10]
```diff
@@ -1,46 +1,127 @@
...
+  } else if (publishType === 'unpublish') {
+    if (publication.track) {
+      publication.track.detach(view);
+    }
+
+    publication.on('subscribed', track => {
+      track.detach(view);
+    });
+  }
+}
```
Comment (manjeshbhargav) [inline examples/screenshare/src/index.js:127]:
```suggestion
}

```

**Review by manjeshbhargav — CHANGES_REQUESTED** [review]

File: examples/screenshare/src/index.js [hunk 11]
```diff
@@ -1,46 +1,121 @@
...
+    name: roomLocal.name,
+    tracks: []
+  });
+
   // Hide the "Stop Capture Screen" button.
   stopScreenCapture.style.display = 'none';
 
   // The LocalVideoTrack for your screen.
   let screenTrack;
+  // let localScreenTrack;
```
Comment (manjeshbhargav) [inline examples/screenshare/src/index.js:41]:
```suggestion
```

File: examples/screenshare/src/index.js [hunk 12]
```diff
@@ -1,46 +1,121 @@
...
   };
 
-  stopScreenCapture.onclick = function() {
-    // Stop capturing your screen.
+  // Stop capturing your screen.
+  const stopScreenSharing = () => {
     screenTrack.stop();
+  };
+
+  stopScreenCapture.onclick = stopScreenSharing;
```
Comment (manjeshbhargav) [inline examples/screenshare/src/index.js:72]:
```suggestion
  // Stop capturing your screen.
  stopScreenCapture.onclick = () => screenTrack.stop();
```

**Review by manjeshbhargav — APPROVED** [review]
